### ISSUE TITLE
Bug: [prefer-nullish-coalescing] ignoreBooleanCoercion should not apply to ternary expressions

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#11581]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3lAXPTMAligObwA%2B8GMArgpTSqAGbEjADcAUKJLAsnRYARngLEylYUiQQQUFBXiMWbTly6DM8AB7wAvPABCMuQoAUUeAH5s8PMICUHIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6MAG1w2HAHdJ0JpAA0a9VnXZIleU3GcAQuPFJJTAMLjec8boz44iQ9gC%2BfgC6aoH%2BQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const a: string | true | undefined;
declare const b: string | boolean | undefined;

const x = Boolean(a ? a : b);
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/prefer-nullish-coalescing": [
      "warn",
      {
        "ignoreBooleanCoercion": true
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

ternary _should_ be reported

### Actual Result

ternary is not reported

### Additional Info

The motivation for `ignoreBooleanCoercion` was that it's helpful sometimes to write `Boolean(a || b || c || d)` rather than being forced into `Boolean(a) || Boolean(b) || Boolean(c) || Boolean(d)` (see https://github.com/typescript-eslint/typescript-eslint/issues/9080). I don't think that that is true of `Boolean(a ? a : b ? b : c ? c : d)`
